test/extended/builds/ldap_fixtures.go

```diff
@@ -0,0 +1,90 @@
+package builds
+
+import (
+	"fmt"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = Describe("default: OpenLDAP build and deployment", func() {
+	defer GinkgoRecover()
+	var (
+		baseImageStreamName      = "openldap"
+		imageStreamFixture       = exutil.FixturePath("fixtures", "ldap", "ldapserver-imagestream.json")
+		imageStreamTargetFixture = exutil.FixturePath("fixtures", "ldap", "ldapserver-imagestream-testenv.json")
+		buildConfigFixture       = exutil.FixturePath("fixtures", "ldap", "ldapserver-buildconfig.json")
+		deploymentConfigFixture  = exutil.FixturePath("fixtures", "ldap", "ldapserver-deploymentconfig.json")
+		serviceConfigFixture     = exutil.FixturePath("fixtures", "ldap", "ldapserver-service.json")
+		oc                       = exutil.NewCLI("openldap", exutil.KubeConfigPath())
+	)
+
+	Describe("Building and deploying an OpenLDAP server", func() {
+		It(fmt.Sprintf("should create a image from %q template and run it in a pod", buildConfigFixture), func() {
+			oc.SetOutputDir(exutil.TestContext.OutputDir)
+
+			By(fmt.Sprintf("calling oc create -f %q", imageStreamFixture))
```

Comment:
Call create on the entire directory at once, then wait for each condition to finish.